src/openrct2/world/ConstructionClearance.cpp

```diff
@@ -152,6 +152,17 @@ GameActions::Result MapCanConstructWithClearAt(
             if (pos.baseZ < tileElement->GetClearanceZ() && pos.clearanceZ > tileElement->GetBaseZ()
                 && !(tileElement->IsGhost()))
             {
+                if (tileElement->GetType() == TileElementType::SmallScenery)
+                {
+                    auto isDiagonal = quarterTile.GetBaseQuarterOccupied();
```

Comment:
isDiagonal implies a bool. Additionally, it’s not immediately clear what the numbers 5 and 10 mean.

```suggestion
                    auto occupiedQuarters = quarterTile.GetBaseQuarterOccupied();
                    auto isDiagonal = (occupiedQuarters == 0b0101 || occupiedQuarters == 0b1010);
```
and modify the check accordingly.